src/app/delegate/api/register/route.ts

```diff
@@ -68,166 +51,53 @@ export async function POST(request: Request) {
       signature,
     };
 
-    if (!(await isAddressAnAccount(address))) {
-      return new Response('Address is not an account', {
-        status: 400,
-      });
-    }
-    const sigverification = await verifySigner(body);
-    if (!sigverification) {
-      return new Response('Signature does not match data', {
-        status: 401,
-      });
+    const validationResult = await validateRegistrationRequest(registrationRequest);
+
+    if (validationResult.length) {
+      return wrapResponseInJson(
+        {
+          status: RegisterDelegateResponseStatus.Error,
+          message: 'Invalid delegatee registration request',
+          errors: validationResult,
+        },
+        400,
```

Comment:
I say that having specific status codes. (401, 400, with precise meanings) is better than only using 1 status code)

Reply:
we talked about IRL and decided to not change it